File: Swift/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.swift
import Foundation

func solution(_ babbling:[String]) -> Int {
    let canSpeakWords = ["aya", "ye", "woo", "ma"]
    var count = 0
    
    for word in babbling {
        var tempWord = word
        var possible = true
        
        for speakWord in canSpeakWords {
            if tempWord.contains(speakWord + speakWord) {
                possible = false
                break
            }
        }
        
        guard possible else { continue }
        
        for speakWord in canSpeakWords {
            tempWord = tempWord.replacingOccurrences(of: speakWord, with: " ")
        }
        
        if tempWord.trimmingCharacters(in: .whitespaces).isEmpty {
            count += 1
        }
    }
    
    return count
}

File: Java/프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        String [] canSpeak = new String[]{"aya", "ye", "woo", "ma" };
        int answer = 0;
        
        for (String word : babbling) {
            boolean possible = true;
            
            for (String speakWord : canSpeak) {
                if (word.contains(speakWord+speakWord)) {
                    possible = false;
                    break;
                }
            }
            
            if (!possible) { continue; }
            
            for (String speakWord : canSpeak) {
                word = word.replaceAll(speakWord, " ");
            }
            
            if (word.trim().equals("")) {
                answer++;
            }
            
        }
        
        return answer;
    }
}